docs/design/storage-auto-scale.md

```diff
@@ -72,60 +72,80 @@ Controller:
 
 1) Create a named controller that watches for [channel generic event](https://book-v1.book.kubebuilder.io/beyond_basics/controller_watches) per device class.
 
-2) If the expansion is not in progress set the status `phase` to `NotStarted` 
+2) Set the status `phase` to `NotStarted`, if no expansion has triggered(status.phase==""). 
+   
+3) Check if the expansion is in progress:
+   
+   1) Query actualOsdCount and actualOsdSize from the from Prometheus .
+   
+   2) Calculate the desiredOsdCount and desiredOsdSize from the Storagecluster.
+   
+   3) If the (actualOsdSize < desiredOsdSize || actualOsdCount < desiredOsdCount), expansion is in progress. 
+   
+4) If the expansion is in progress, check the progress and then requeue each 1 minute until the expansion is completed successfully, change the phase to `failed` if scaling not `Succeeded` with in timeoutSeconds(default:1800) interval.
    
-3) If an expansion is in progress(expectedOsdSize!=startOsdSize || expectedOsdCount!=startOsdCount), check the progress and then requeue each 1 minute until the expansion is completed successfully(jump to step 11).
+5) If no-expansion is in progress,
    
-4) If the LSO storageclass is detected in the storageClassDeviceSet, raise a warning and do not recocnile further.
+   1) Check the (status.phase == "InProgress"), if yes scaling is successful, update the status of the `StorageAutoScaling` CR with `lastExpansionCompletionTime` and `phase` and also osd count and size.
```

Comment:
It's confusing that it would be InProgress if it's detected that it's not in progress. But if this code path is just part of the status check, then I'm sure it's fine.

Reply:
It's just for updating the status to success from in progress, when we have already detected it was succeeded